cpp/src/arrow/engine/substrait/extension_set.h

```diff
@@ -408,6 +412,13 @@ class ARROW_ENGINE_EXPORT ExtensionSet {
   /// \return An anchor that can be used to refer to the type within a plan
   Result<uint32_t> EncodeType(const DataType& type);
 
+  /// \brief Lookup the anchor for a given type alias
+  ///
+  /// Similar to \see EncodeType but this is used for cases where the data type is either
+  /// parameterized or custom in some way (e.g. we use this for Time64::Nanos).  We need
+  /// to use the Id directly since we can't have registered the type with the registry.
+  Result<uint32_t> EncodeTypeId(Id type_id);
```

Comment:
AFAICT this isn't used anywhere. If let's wait to add it until it's necessary

Reply:
Good catch, that was from an aborted attempt to add support for parameterized UDTs.  I've removed.